WireAuthentication/Sources/WireAuthenticationUI/Views/Root/RootView.swift

```diff
@@ -43,12 +47,20 @@ package struct RootView: View {
     }
 
     @ViewBuilder
-    private func sheetContent(for sheet: RootView.ModalDestination) -> some View {
+    private func sheetContent(for sheet: RootViewSheet) -> some View {
         switch sheet {
-        case let .authFlow(backedInfo):
+        case let .authFlow(backendInfo):
             NavigationStack(path: $viewModel.path) {
-                factory.determineAuthMethodView(backendInfo: backedInfo)
+                DetermineAuthMethodView(
+                    factory: viewModel.factory.determineAuthMethodFactory(
+                        backendInfo: backendInfo
+                    )
+                )
             }
+            // We must provide an explicit id so it knows to create a new
+            // view when the backend info changes.
+            .id(backendInfo)
```

Comment:
I wonder if we should do this for all views that have view models. What do you think? And also, instead of using the backend info, just set the hashable destination as the id.

Reply:
as of now it was not needed